parser/lexer.go

```diff
@@ -742,7 +742,7 @@ func (l *protoLex) skipToEndOfBlockComment(lval *protoSymType) (ok, hasErr bool)
 func (l *protoLex) addSourceError(err error) (reporter.ErrorWithPos, bool) {
 	ewp, ok := err.(reporter.ErrorWithPos)
 	if !ok {
-		ewp = reporter.Error(l.prev(), err)
+		ewp = reporter.Error(l.prev().AsSpan(), err)
```

Comment:
I think this line (and the one below, line 760) are the only places using `AsSpan`, other than for unknown positions. So these could change to use `ast.NewSourceSpan`. And I think it's worth adding TODOs so that, if possible, these actually capture the span of the relevant token instead of just the single position. (It's definitely possible here; might need more investigation about whether it's do-able for the one below.)

Reply:
done